File: egghead.io/data-structures-and-algorithms-in-js/lesson-13-divide-and-recurse-over-an-arr-with-merge-sort-in-js/merge-sort.js
const mergeSort = (array) => {
  if (array.length < 2) return array;

  const cent = Math.floor(array.length / 2);
  const pred = array.slice(0, cent);
  const succ = array.slice(cent);

  return merge(mergeSort(pred), mergeSort(succ));
};

const merge = (pred, succ) => {
  const sorted = [];

  while (pred.length && succ.length) {
    sorted.push(pred[0] <= succ[0] 
      ? pred.shift() 
      : succ.shift()
    );
  }

  const merged = [...sorted, ...pred, ...succ];
  console.log(merged);
  
  return merged;
};

module.exports = mergeSort;

File: egghead.io/data-structures-and-algorithms-in-js/lesson-13-divide-and-recurse-over-an-arr-with-merge-sort-in-js/merge_sort.py
import math


def merge_sort(array):
  if len(array) < 2: return array
  
  cent = math.floor(len(array) / 2)
  pred, succ = array[:cent], array[cent:]
  
  return merge(merge_sort(pred), merge_sort(succ))

def merge(pred, succ):
  sorted_array = []

  while len(pred) and len(succ):
    sorted_array.append(pred.pop(0) 
      if pred[0] <= succ[0] 
      else succ.pop(0)
    )

  merged_array = sorted_array + pred + succ
  print(merged_array)

  return merged_array